Products/PluggableAuthService/plugins/ZODBGroupManager.py

```diff
@@ -284,6 +303,8 @@ def addPrincipalToGroup( self, principal_id, group_id ):
         if not already:
             new = current + ( group_id, )
             self._principal_groups[ principal_id ] = new
+            notify(PrincipalAddedToGroup(self._get_principal(principal_id),
+                   self._get_group(group_id)))
```

Comment:
In the proposed implementation `PrincipalAddedToGroup.group` and `PrincipalAddedToGroup.principal` can either be a string or an object.

What about removing `_get_principal` and `_get_group` and notify:
```
PrincipalAddedToGroup(self, principal_id, group_id)
```
or 
```
PrincipalAddedToGroup(self._getPAS(), principal_id, group_id)
```
and delegate to the event handler the decision to resolve or not the principal and the group?
It might be faster and more reliable because you will always know that the notified event will have a `principal_id` and a `group_id`.

Reply:
I agree, those two methods are weird and should go.